apps/asap-server/src/controllers/teams.ts

```diff
@@ -67,6 +68,22 @@ export default class Teams implements TeamController {
     return this.fetchById(id);
   }
 
+  async merge(id: string, outputs: string[]): Promise<TeamResponse> {
```

Comment:
I wonder if our team controller should really expose such method - what if we change the underlying data store and the relationship can be embedded on the user input? I think I would simply add this logic to the `create` method instead, so that it's all done (or could be done) atomically

Reply:
btw you could achieve the 'atomicity' (thats a word apparently) then by inserting the RO, then try/catching the team merging and if it fails - removing the RO

Reply:
We are trying to do 2 updates/inserts to squidex.

- Create a RO.
- update a team with the RO relationship.

Ideally we'd like to do this in a transaction, but we can't. 

Say we insert an RO, and then to create the team relationship. If the latter fails, what should happen? We could follow something like the Saga pattern, and try to roll back the previous create (i.e. delete the RO). 

But what happens if this delete/removal fails too? We'd need to build in some retry logic etc. We decided in Refinement to log this error (in another ticket) and we'd have manual intervention for now.

Reply:
> I wonder if our team controller should really expose such method - what if we change the underlying data store and the relationship can be embedded on the user input?

Are you suggesting we move the RO relationship to a team, away from the team record and put it on the RO? Sounds heavy?

Reply:
with regards to the transaction - sure that can be done in a separate ticket so dont worry about it now.


But I still stand by what i said -  think logically it would make more sense to create a RO through a single controller method, i dont think we should expose a 'merge' method from the team controller. To me all the controllers (or alter - the clients/repositories if we shift the db logic further down) should only expose methods like create/update/delete.